pkg/scrapper/scrapper.go

```diff
@@ -0,0 +1,122 @@
+package scrapper
+
+import (
+	"fmt"
+	"strings"
+	"IV-GHL/internal"
+	"regexp"
+)
+
+func ExtraerLinea(htmlContent string) (*internal.Linea, error) {
+	linea := &internal.Linea{
+		TipoMedio:              internal.AUTOBUS,
+		Horario_Paradas_Ida:    make(map[string][]string),
+		Horario_Paradas_Vuelta: make(map[string][]string),
+	}
+
+	// Extracción número de la línea
+	numLinea, error := GetNumLinea(htmlContent)
+	if(error != nil){
+		return linea, error
+	}
+
+	linea.NumLinea = numLinea
+
+	// Bloques ida y vuelta
+	idxIda := strings.Index(htmlContent, "<strong>Ida</strong>")
+	idxVuelta := strings.Index(htmlContent, "<strong>Vuelta</strong>")
+	idxFinVuelta := strings.Index(htmlContent, `<div class="leyendas">`)
+
+	if idxIda == -1 {
+		error = fmt.Errorf("no se encontraron los marcadores Ida")
+		return linea, error
+	}
+	if idxVuelta == -1 {
+		error = fmt.Errorf("no se encontraron los marcadores Vuelta")
+		return linea, error
+	}
+	if idxFinVuelta == -1 {
+		error = fmt.Errorf("no se encontraron los marcadores FinVuelta")
+		return linea, error
+	}
+
+	bloqueIda := htmlContent[idxIda:idxVuelta]
+	bloqueVuelta := htmlContent[idxVuelta:idxFinVuelta]
+
+	// Extracción de nombres de paradas y horarios de ida y vuelta
+	nombresIda := ExtraerNombresParadas(bloqueIda)
+	for i := 0; i < len(nombresIda); i++ {
+		linea.Horario_Paradas_Ida[nombresIda[i]] = append(linea.Horario_Paradas_Ida[nombresIda[i]], "")
+	}
+
+	nombresVuelta := ExtraerNombresParadas(bloqueVuelta)
+	for i := 0; i < len(nombresVuelta); i++ {
+		linea.Horario_Paradas_Vuelta[nombresVuelta[i]] = append(linea.Horario_Paradas_Vuelta[nombresVuelta[i]], "")
+	}
+
+	return linea, error
+}
+
+func GetNumLinea(htmlContent string) (string, error) {
+	// <h2> : Etiqueta de apertura
+    // \s* : Salta espacios en blanco iniciales
+    // (.*?) : Captura todos los caracteres que encuentra hasta el cierre de etiqueta, especificado detrás.
+    // \s* : Salta espacios en blanco finales
+    // </h2> : Etiqueta de cierre
+    re := regexp.MustCompile(`<h2>\s*(.*?)\s*</h2>`)
+    match := re.FindStringSubmatch(htmlContent)
+
+    if len(match) < 2 {
+        return "", fmt.Errorf("el contenido HTML no contiene el encabezado <h2> necesario para el título")
```

Comment:
No has modelizado los errores. Mira [esto](https://stackoverflow.com/questions/56580030/can-we-create-subtypes-for-errors-in-go) para una posible solución. [Aquí](https://pkg.go.dev/errors) habla de un paquete que lo hace y finalmente en [estos ejemplos](https://gobyexample.com/errors) explica cómo se definen errores específicos en Go.